src/components/create-vm-dialog/createVmDialog.jsx

```diff
@@ -748,20 +752,98 @@ const UsersConfigurationRow = ({
     );
 };
 
+// This method needs to be outside of component as re-render would create a new instance of debounce
+const parseKey = debounce(500, (key, setKeyObject) => {
+    if (isEmpty(key))
+        return;
+
+    // Validate correctness of the key
+    return cockpit.spawn(["ssh-keygen", "-l", "-f", "-"], { error: "message" })
+            .input(key)
+            .then(() => {
+                const parts = key.split(" ");
+                if (parts.length > 2) {
+                    setKeyObject({
+                        type: parts[0],
+                        data: parts[1],
+                        comment: parts[2], // comment is optional in SSH-format
+                    });
+                }
+            })
+            .catch(() => {
+                setKeyObject(undefined);
+                console.warn("Could not validate the public key");
+            });
+});
+
+const SshKeysRow = ({
+    id, item, onChange, idx, removeitem,
+}) => {
+    const [keyObject, setKeyObject] = useState();
+
+    const onChangeHelper = (value) => {
+        // Some users might want to input multiple keys into the same field
+        // While handling that in the future might be a nice user experience, now we only parse one key out of input
+        value = value.split(/\r?\n/)[0];
+
+        onChange(idx, "value", value);
+        parseKey(value, setKeyObject);
+    };
+
+    return (
+        <Grid id={id} key={id}>
+            <GridItem span={11}>
+                {keyObject
+                    ? <FlexItem id="validated">
+                        <strong>{keyObject.comment}</strong>
+                        <span>{keyObject.comment ? " - " + keyObject.type : keyObject.type}</span>
+                        <div>{keyObject.data}</div>
+                    </FlexItem>
+                    : <FormGroup label={_("Public key")}
+                        fieldId='public-key'>
+                        <TextArea value={item.value || ""}
+                            aria-label={_("Public SSH key")}
+                            onChange={(_, value) => onChangeHelper(value)}
+                            rows="3" />
+                        <FormHelper helperText={_("Keys are located in ~/.ssh/ and have a \".pub\" extension.")} />
+                    </FormGroup>
+                }
+            </GridItem>
+            <GridItem span={1} className="pf-m-1-col-on-md remove-button-group">
+                <Button variant='plain'
+                    className="btn-close"
+                    id={id + "-btn-close"}
+                    isSmall
+                    aria-label={_("Remove item")}
+                    icon={<TrashIcon />}
+                    onClick={() => removeitem(idx)} />
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1150-20231009-100708-56b4bdba-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)